File: coderise.service/Models/ShelvesetModel.cs
using System;

namespace coderise.service.Models
{
    public class ShelvesetModel
    {
        public long Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Owner { get; set; }

        public DateTime CreatedAt { get; set; }

        public string[] FileList { get; set; }
    }
}

File: coderise.web/js/models/shelvesetModel.js
var ShelvesetModel = Backbone.Model.extend({
    validate: function (attrs) {
        if (!attrs.id) {
            return 'Shelveset id is required';
        }

        if (!attrs.title) {
            return 'Shelveset title is required';
        }

        if (!attrs.owner) {
            return 'Shelveset owner is required';
        }

        if (!attrs.createdAt) {
            return 'Shelveset creation date is required';
        }

        if (!attrs.fileList) {
            return 'Shelveset must contain not empty file list';
        }
        return "";
    }
});